packages/server/lib/controllers/proxy.controller.ts

```diff
@@ -0,0 +1,320 @@
+
+            const { account_id: accountId, provider_config_key: providerConfigKey, connection_id: connectionId } = connection as Connection;
+            const providerConfig = await configService.getProviderConfig(providerConfigKey, accountId);
+
+            if (!providerConfig) {
+                logger.error(`Proxy: provider configuration not found`);
+                res.status(404).send();
+            }
+            const template = configService.getTemplate(String(providerConfig?.provider));
+
+            if (!template.base_api_url) {
+                logger.error(`Proxy: base api url configuration is missing. The providers.yaml might be missing a value`);
+                errorManager.errRes(res, 'missing_base_api_url');
+                return;
+            }
+
+            logger.info(`Proxy: API call configuration constructed successfully`);
+
+            const configBody: ProxyBodyConfiguration = {
+                endpoint,
+                method: method as HTTP_VERB,
+                template,
+                // handle oauth1
+                token: String(token),
+                providerConfigKey,
+                connectionId,
+                data: req.body
+            };
+
+            return this.sendToHttpMethod(res, next, method as HTTP_VERB, configBody);
+        } catch (error) {
+            next(error);
+        }
+    }
+
+    private async get(res: Response, next: NextFunction, url: string, config: ProxyBodyConfiguration) {
+        try {
+            const headers = this.constructHeaders(config);
+            const responseStream: AxiosResponse = await axios({
+                method: 'get',
+                url,
+                responseType: 'stream',
+                headers
+            });
+            logger.info(`Proxy: GET request to ${url} was successful`);
+            res.writeHead(responseStream?.status, responseStream.headers as OutgoingHttpHeaders);
+            responseStream.data.pipe(res);
+        } catch (error) {
+            const nangoError = this.catalogAndReportError(error as Error | AxiosError, url);
+            next(nangoError);
+        }
+    }
+
+    private sendToHttpMethod(res: Response, next: NextFunction, method: HTTP_VERB, configBody: ProxyBodyConfiguration) {
+        const url = this.constructUrl(configBody);
+
+        if (method === 'POST') {
+            return this.post(res, next, url, configBody);
+        } else if (method === 'PATCH') {
+            return this.patch(res, next, url, configBody);
+        } else if (method === 'PUT') {
+            return this.put(res, next, url, configBody);
+        } else if (method === 'DELETE') {
+            return this.delete(res, next, url, configBody);
+        } else {
+            return this.get(res, next, url, configBody);
+        }
+    }
+
+    private async post(res: Response, next: NextFunction, url: string, config: ProxyBodyConfiguration) {
+        try {
+            if (!config.data) {
```

Comment:
Could we have a POST request without body? It looks like it's okay: https://stackoverflow.com/questions/4191593/is-it-considered-bad-practice-to-perform-http-post-without-entity-body

Reply:
Thanks, updated with https://github.com/NangoHQ/nango/pull/562/commits/a9e3dab846a3658fd02979f33f8f85687094d3c4